src/python/pants/backend/jvm/tasks/scala_rewrite_base.py

```diff
@@ -19,11 +19,13 @@
 class ScalaRewriteBase(NailgunTask, AbstractClass):
   """Abstract base class for both scalafix and scalafmt: tools that check/rewrite scala sources."""
 
+  _TARGET_TYPES = ['scala_library', 'junit_tests', 'java_tests']
```

Comment:
Here we make the minimally invasive change. What do you think about renaming this file `rewrite_base.py` and removing the Scala references? This class seems generally relevant to any JVM-based formatter/linter tool, rather than being tied to the language of sources being formatted/linted.

Reply:
I like it. I'd prefer a manually abstract classmethod though that doc could be hung off of, ie:
```python
@classmethod
def default_target_types(cls):
  """Subclasses should ..."""
  raise NotImplementedError()
```

That said, this is the existing style, so take it or leave it. I just went down the rabbit hole of checking `_SCALA_SOURCE_EXTENSION` - another example of this - and I find it mystifying as-is since its not defined in the subclass, not documented, but required to be defined by leaf subclasses in order to not blow up.